src/libsync/propagatedownload.cpp

```diff
@@ -672,8 +669,27 @@ void PropagateDownloadFile::startDownload()
 
     // Can't open(Append) read-only files, make sure to make
     // file writable if it exists.
-    if (_tmpFile.exists())
+    if (_tmpFile.exists()) {
         FileSystem::setFileReadOnly(_tmpFile.fileName(), false);
+    }
+
+    try {
+        const auto newDirPath = std::filesystem::path{_tmpFile.fileName().toStdWString()};
+        Q_ASSERT(newDirPath.has_parent_path());
+        _parentFolderPath = newDirPath.parent_path();
+        auto parentFolderStatus = std::filesystem::status(_parentFolderPath);
+        _parentPermissions = parentFolderStatus.permissions();
+        if ((_parentPermissions & std::filesystem::perms::owner_write) != std::filesystem::perms::owner_write) {
```

Comment:
@mgallien this looks very similar to what you've added to `src/libsync/propagatorjobs.cpp`. Can this be factored out with a descriptive name, and used in both classes? Maybe, you could return `bool` and then call `emit propagator()->touchedFile(...)` if needed?